### PR TITLE
Add scheduler to remove temp files

### PR DESCRIPTION
**Pull request by fefrain**

### CODE REVIEW

File: server/system/schedulers/recordPreviewCleanup.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const schedule = require('node-schedule')
 
-const SurveyManager = require('../modules/survey/manager/surveyManager')
-const RecordService = require('../modules/record/service/recordService')
+const SurveyManager = require('../../modules/survey/manager/surveyManager')
+const RecordService = require('../../modules/record/service/recordService')
```
Comment (minotogna) [inline server/system/schedulers/recordPreviewCleanup.js]: Can we use jus RecordService here ? (i know it was already there)

File: server/system/schedulers/tempFilesCleanup.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  try {
+    const files = await fs.readdirSync(tempFolder)
+    for (const file of files) {
+      const filePath = path.join(tempFolder, file)
+      const stat = await fs.statSync(filePath)
+      if (stat.isFile())
+        await visitor(filePath, stat)
+    }
+  } catch (error) {
+    //ignore errors (eg. temp folder not created yet, unable to delete temp file, etc)
```
Comment (minotogna) [inline server/system/schedulers/tempFilesCleanup.js]: Can you log the error?

File: server/system/schedulers/tempFilesCleanup.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    // cleanup temp files older than 6 hours
+    await cleanupTempFiles(6)
+  })
+
+const cleanupTempFiles = async (olderThanHours = null) => {
+  Logger.debug('Cleaning up temp files')
+
+  let count = 0
+  const now = new Date()
+
```
Comment (minotogna) [inline server/system/schedulers/tempFilesCleanup.js]: I don't really see the need to call the function visit temp files with the visitor to that just delete the file. Why not to do everything in this method?

Comment (fefrain) [reply server/system/schedulers/tempFilesCleanup.js]: ok

File: server/system/schedulers/tempFilesCleanup.js (hunk 2 above)
Comment (fefrain) [reply server/system/schedulers/tempFilesCleanup.js]: ok

File: server/system/schedulers/recordPreviewCleanup.js (hunk 1 above)
Comment (fefrain) [reply server/system/schedulers/recordPreviewCleanup.js]: right

File: server/system/appCluster.js [hunk 4]
```diff
@@ -66,4 +67,5 @@ module.exports = async () => {
   // ====== schedulers
   await RecordPreviewCleanup.init()
   await ExpiredJwtTokensCleanup.init()
+  await TempFilesCleanup.init()
 }
```
Comment (sgiaccio) [inline server/system/appCluster.js]: Why don't we put the path to the tempFolder in .env and initialize goth the logger and the TempFilesCleanup with its value?

Comment (fefrain) [reply server/system/appCluster.js]: very good idea, thanks!

Comment (minotogna) [reply server/system/appCluster.js]: Cool. Please update the read me as well

File: server/system/environment.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  tempFolder: 'TEMP_FOLDER'
+}
+
+const defaultTempFolder = '/tmp/arena_upload'
+
+const tempFolder = R.propOr(defaultTempFolder, keys.tempFolder)(process.env)
+
+module.exports = {
+  tempFolder
+}
```
Comment (minotogna) [inline server/system/environment.js]: what about adding to common/processUtils and ENV object that contains all the env variables? this mean all references to process.env should be replace by ```ProcessUtils.ENV.[var]```?

File: server/system/environment.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const R = require('ramda')
+
+const keys = {
+  tempFolder: 'TEMP_FOLDER'
+}
+
+const defaultTempFolder = '/tmp/arena_upload'
+
+const tempFolder = R.propOr(defaultTempFolder, keys.tempFolder)(process.env)
```
Comment (minotogna) [inline server/system/environment.js]: defaultTempFolder can be inline

File: server/system/environment.js (hunk 5 above)
Comment (fefrain) [reply server/system/environment.js]: honestly I don't like the "utils" prefix, and I would prefer to have a getter for every environment variable

File: server/system/environment.js (hunk 6 above)
Comment (fefrain) [reply server/system/environment.js]: ok

File: server/system/environment.js (hunk 5 above)
Comment (minotogna) [reply server/system/environment.js]: why a getter ?

File: server/utils/mailer.js [hunk 7]
```diff
@@ -7,7 +8,7 @@ const sendEmail = async (to, msgKey, msgParams = {}, lang) => {
   const mailSubject = i18n.t(`${msgKey}.subject`)
   const msgBody = i18n.t(`${msgKey}.body`, msgParams)
 
-  await aws.sendEmail(process.env.ADMIN_EMAIL, to, mailSubject, msgBody)
+  await aws.sendEmail(ProcessUtils.ENV.adminEmail, to, mailSubject, msgBody)
```
Comment (minotogna) [inline server/utils/mailer.js]: do you mean ProcessEnv.adminEmail?

Comment (minotogna) [reply server/utils/mailer.js]: I also didn't see that we need to specify ADMIN_EMAIL in .env @sgiaccio and neither we were notified. @fefrain could you add it to the README, thanks.

File: server/system/processEnv.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const ProcessUtils = require('../../common/processUtils')
+
```
Comment (minotogna) [inline server/system/processEnv.js]: to me ProcessEnv should go under server/utils. It's not used only in system, but rather into several server modules

File: server/utils/mailer.js (hunk 7 above)
Comment (fefrain) [reply server/utils/mailer.js]: ok

File: server/system/processEnv.js (hunk 8 above)
Comment (fefrain) [reply server/system/processEnv.js]: ok

File: webpack.config.js [hunk 9]
```diff
@@ -9,15 +9,17 @@ const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPl
...
+const ProcessUtils = require('./common/processUtils')
+
 const mode = {
   development: 'development',
   production: 'production'
 }
 
-const prodBuild = process.env.NODE_ENV === mode.production
-const buildReport = process.env.BUILD_REPORT === 'true'
+const prodBuild = ProcessUtils.ENV.nodeEnv === mode.production
```
Comment (minotogna) [inline webpack.config.js]: can we add ProcessUtils.envProduction ? (internal mode object should be removed)

**Pull request merged**